mod_form.php

```diff
@@ -292,4 +300,24 @@ public function add_completion_rules() {
     public function completion_rule_enabled($data) {
         return true;
     }
+    public function validation($data, $files) {
+        $errors = parent::validation($data, $files);
+
+        if (!empty($data['emailperiod']) && $data['emailperiod'] <= 60 && $data['emailfrequency'] > 5) {
+            // Email is sent based on minutes selector - make sure no more than 5 emails can be sent.
+            $errors['emailfrequency'] = get_string('frequencytoohigh', 'reengagement', 5);
+        } else if (!empty($data['emailperiod']) && $data['emailperiod'] <= 3600 && $data['emailfrequency'] > 12) {
+            // Email is sent based on hours selector - makes sure no more than 12 emails can be sent.
+            $errors['emailfrequency'] = get_string('frequencytoohigh', 'reengagement', 12);
+        } else if (!empty($data['emailperiod']) && $data['emailperiod'] <= 86400 && $data['emailfrequency'] > 24) {
+            // Email is sent based on days selctor - make sure no more than  24 emails can be sent.
+            $errors['emailfrequency'] = get_string('frequencytoohigh', 'reengagement', 24);
+        }
+
+        if ($data['emailperiod'] == 60 && $data['emailperiodcount'] < 5) {
+            $errors['emaildelay'] = get_string('periodtoolow', 'reengagement');
```

Comment:
yes - but UI is not great - it just makes that change "silently" - it should really notify the user during validation right?